src/utils/layout/generator/GeneratorDataSources.tsx

```diff
@@ -0,0 +1,147 @@
+import { createHookContext } from 'src/core/contexts/hookContext';
+import { useApplicationMetadata } from 'src/features/applicationMetadata/ApplicationMetadataProvider';
+import { useApplicationSettings } from 'src/features/applicationSettings/ApplicationSettingsProvider';
+import { DataModels } from 'src/features/datamodel/DataModelsProvider';
+import { useDevToolsStore } from 'src/features/devtools/data/DevToolsStore';
+import { useExternalApis } from 'src/features/externalApi/useExternalApi';
+import { useLayoutSets } from 'src/features/form/layoutSets/LayoutSetsProvider';
+import { useCurrentLayoutSet } from 'src/features/form/layoutSets/useCurrentLayoutSet';
+import { FD } from 'src/features/formData/FormDataWrite';
+import { useLaxDataElementsSelectorProps, useLaxInstanceDataSources } from 'src/features/instance/InstanceContext';
+import { useLaxProcessData } from 'src/features/instance/ProcessContext';
+import { useCurrentLanguage } from 'src/features/language/LanguageProvider';
+import { useInnerLanguageWithForcedNodeSelector } from 'src/features/language/useLanguage';
+import { Validation } from 'src/features/validation/validationContext';
+import { useMultipleDelayedSelectors } from 'src/hooks/delayedSelectors';
+import { useShallowObjectMemo } from 'src/hooks/useShallowObjectMemo';
+import { useCommitWhenFinished } from 'src/utils/layout/generator/CommitQueue';
+import { Hidden, NodesInternal, useNodesWhenReadyOrInit } from 'src/utils/layout/NodesContext';
+import { useInnerDataModelBindingTranspose } from 'src/utils/layout/useDataModelBindingTranspose';
+import { useInnerNodeFormDataSelector } from 'src/utils/layout/useNodeItem';
+import { useInnerNodeTraversalSelector } from 'src/utils/layout/useNodeTraversal';
+import type { ValidationDataSources } from 'src/features/validation';
+import type { ExpressionDataSources } from 'src/utils/layout/useExpressionDataSources';
+
+const { Provider, hooks } = createHookContext({
+  useLaxInstanceDataSources: () => useLaxInstanceDataSources(),
+  useCurrentLayoutSet: () => useCurrentLayoutSet(),
+  useDefaultDataType: () => DataModels.useDefaultDataType(),
+  useReadableDataTypes: () => DataModels.useReadableDataTypes(),
+  useExternalApis: () => useExternalApis(useApplicationMetadata().externalApiIds ?? []),
+  useNodes: () => useNodesWhenReadyOrInit(),
+  useIsForcedVisibleByDevTools: () => {
+    const devToolsIsOpen = useDevToolsStore((state) => state.isOpen);
+    const devToolsHiddenComponents = useDevToolsStore((state) => state.hiddenComponents);
+    return devToolsIsOpen && devToolsHiddenComponents !== 'hide';
```

Comment:
Not that it matters all that much, but
```suggestion
    return useDevToolsStore((state) => state.isOpen && state.hiddenComponents !== 'hide');
```